rootston/wl_shell.c

```diff
@@ -243,7 +243,7 @@ void handle_wl_shell_surface(struct wl_listener *listener, void *data) {
 	view->close = close;
 	roots_surface->view = view;
 	view_init(view, desktop);
-	wl_list_insert(&desktop->views, &view->link);
+	desktop_add_view(desktop, view);
```

Comment:
```
desktop_add_view
view_setup
```

I would look at this first to get the insertion right.

Reply:
Hmm, good point, hadn't noticed `view_setup` because I was mostly looking at removal but that will explain what I said about insertion. I notice this is calling `view_update_output`, that's also called on view move, if I can do something a bit less naive I'll change `desktop_add_view` to `view_update_cursors`.

Obviously I can't reproduce the problem I seemed to have with removing a view yesterday, so I guess I'll start with that...
For the record, we also get "fun" behaviour if a resize is active (logo+right click or edge drag) and alt-tab happens. It's not as bad as I had expected but I'm not sure how we'll want to handle that, fun stuff.